### PR TITLE
Make PickingManager thread safe and simplify interface.

### PR DESCRIPTION
**Pull request by pierricgimmig**
This was started as an investigation into a rare crash that happens on picking, although I couldn't repro the crash after the changes, I'm still not 100% sure it is fixed.  I'll keep an eye on it.  Making PickingManager thread safe is needed for the upcoming parallelization of rendering.

### CODE REVIEW

File: OrbitGl/PickingManager.cpp [hunk 1]
```diff
@@ -8,52 +8,77 @@
 
 //-----------------------------------------------------------------------------
 PickingID PickingManager::CreatePickableId(Pickable* a_Pickable) {
+  absl::MutexLock lock(&mutex_);
   ++m_IdCounter;
-  PickingID id = PickingID::Get(PickingID::PICKABLE, ++m_IdCounter);
```
Comment (ronaldfw) [inline OrbitGl/PickingManager.cpp:12]: Oh, was this incremented twice here by accident?

File: OrbitGl/PickingManager.cpp [hunk 2]
```diff
@@ -8,52 +8,77 @@
...
+  absl::MutexLock lock(&mutex_);
   m_IdPickableMap.clear();
   m_PickableIdMap.clear();
+  m_IdCounter = 0;
+}
+
+//-----------------------------------------------------------------------------
+Pickable* PickingManager::GetPickableFromId(uint32_t id) {
+  absl::MutexLock lock(&mutex_);
+  return m_IdPickableMap[id];
```
Comment (ronaldfw) [inline OrbitGl/PickingManager.cpp:30]: Is it ok if "id" is not yet an existing key here? This adds a new element to the map, is this intended? (If it can never happen with normal usage, I would add a check / assert here that tests if the id exists in the map.)

File: OrbitGl/PickingManager.cpp [hunk 3]
```diff
@@ -8,52 +8,77 @@
...
 void PickingManager::Pick(uint32_t a_Id, int a_X, int a_Y) {
-  m_Picked = m_IdPickableMap[a_Id];
-  if (m_Picked) {
-    m_Picked->OnPick(a_X, a_Y);
+  Pickable* picked = GetPickableFromId(a_Id);
+  if (picked) {
+    picked->OnPick(a_X, a_Y);
   }
+
+  absl::MutexLock lock(&mutex_);
```
Comment (ronaldfw) [inline OrbitGl/PickingManager.cpp:46]: It looks like the lock is only added here because GetPickableFromId also locks. Maybe add a comment about that?

File: OrbitGl/PickingManager.cpp [hunk 4]
```diff
@@ -8,52 +8,77 @@
...
+
+//-----------------------------------------------------------------------------
+Color PickingManager::GetPickableColor(Pickable* pickable) {
+  PickingID id = CreatePickableId(pickable);
+  return ColorFromPickingID(id);
 }
 
 //-----------------------------------------------------------------------------
 Color PickingManager::ColorFromPickingID(PickingID id) const {
-  static_assert(sizeof(PickingID) == sizeof(Color),
```
Comment (ronaldfw) [inline OrbitGl/PickingManager.cpp:55]: Why did you remove this? Is this not a requirement anymore?

File: OrbitGl/PickingManager.cpp (hunk 1 above)
Comment (pierricgimmig) [reply OrbitGl/PickingManager.cpp:12]: Yes :-)

File: OrbitGl/PickingManager.cpp (hunk 2 above)
Comment (pierricgimmig) [reply OrbitGl/PickingManager.cpp:30]: It is actually OK.  In the case where we inadvertently draw primitives using a color that is wasn't generated as a Picking Color, when we ask for the corresponding Pickable, it will return a nullptr.  We should probably warn in that case.  I'll add that.

File: OrbitGl/PickingManager.cpp (hunk 3 above)
Comment (pierricgimmig) [reply OrbitGl/PickingManager.cpp:46]: It locks to protect access to m_Picked. GetPicked() also locks.

File: OrbitGl/PickingManager.cpp (hunk 4 above)
Comment (pierricgimmig) [reply OrbitGl/PickingManager.cpp:55]: It's actually a duplicate check, a similar one is present in the PickingID class.  Maybe I should leave it here for clarity actually, and remove the one in the header.  Will do.

File: OrbitGl/PickingManager.cpp (hunk 2 above)
Comment (pierricgimmig) [reply OrbitGl/PickingManager.cpp:30]: In the end I chose not to warn as this happens any time we click at a location where there is no "pickable".  I did however made it so that we don't add the nullptr to the map but rather return nullptr explicitly.
